File: scripts/migrate-scss.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const { glob } = require('glob');

console.log('🔧 Starting SCSS migration to modern @use syntax...\n');

// Find all .scss files in src directory
const scssPattern = path.join(__dirname, '../monte-da-estrada/src/**/*.scss');
const scssFiles = glob.sync(scssPattern, {
  ignore: ['**/node_modules/**', '**/dist/**']
});

console.log(`Found ${scssFiles.length} SCSS files to migrate\n`);

let migratedCount = 0;

scssFiles.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  let modified = false;
  const originalContent = content;

  // Skip if already using @use for styles
  if (content.includes("@use '@/styles' as *;") || content.includes('@use "./variables"')) {
    console.log(`⏭️  Skipped (already migrated): ${path.relative(process.cwd(), file)}`);
    return;
  }

  // Replace @import '@/styles/variables' and @import '@/styles/mixins' with single @use
  if (content.includes("@import '@/styles/variables'") || content.includes("@import '@/styles/mixins'")) {
    // Remove both imports
    content = content.replace(/@import\s+['"]@\/styles\/variables['"]\s*;\s*/g, '');
    content = content.replace(/@import\s+['"]@\/styles\/mixins['"]\s*;\s*/g, '');

    // Add single @use statement after first comment block
    const lines = content.split('\n');
    let insertIndex = 0;

    // Find end of comment block
    for (let i = 0; i < lines.length; i++) {
      if (lines[i].trim().startsWith('//') || lines[i].trim().startsWith('/*') || lines[i].trim() === '') {
        insertIndex = i + 1;
      } else {
        break;
      }
    }

    // Insert @use statement
    lines.splice(insertIndex, 0, "@use '@/styles' as *;", '');
    content = lines.join('\n');

    // Clean up multiple empty lines
    content = content.replace(/\n\n\n+/g, '\n\n');

    modified = true;
  }

  if (modified) {
    fs.writeFileSync(file, content, 'utf8');
    migratedCount++;
    console.log(`✅ Migrated: ${path.relative(process.cwd(), file)}`);
  }
});

console.log(`\n✨ Migration complete! Updated ${migratedCount} files.`);
console.log('\n📋 Next steps:');
console.log('1. Run: cd monte-da-estrada && npm run dev');
console.log('2. Verify no SCSS errors in console');
console.log('3. Test all pages visually');

File: scripts/migrate-scss.py
#!/usr/bin/env python3

import os
import re
import glob

print("Starting SCSS migration to modern @use syntax...\n")

# Find all .module.scss files
pattern = os.path.join(os.path.dirname(__file__), '../monte-da-estrada/src/**/*.module.scss')
scss_files = glob.glob(pattern, recursive=True)

print(f"Found {len(scss_files)} SCSS module files\n")

migrated_count = 0

for file_path in scss_files:
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    original_content = content

    # Skip if already migrated
    if "@use '@/styles' as *;" in content:
        print(f"-  Skipped (already migrated): {os.path.basename(file_path)}")
        continue

    # Check if file has old imports
    if "@import '@/styles/variables';" in content or "@import '@/styles/mixins';" in content:
        # Remove both old import lines
        content = re.sub(r"@import\s+['\"]@/styles/variables['\"];\s*\n?", '', content)
        content = re.sub(r"@import\s+['\"]@/styles/mixins['\"];\s*\n?", '', content)

        # Find the position to insert @use (after comments)
        lines = content.split('\n')
        insert_index = 0

        # Find end of comment block
        in_comment = False
        for i, line in enumerate(lines):
            stripped = line.strip()
            if stripped.startswith('//') or stripped.startswith('/*'):
                insert_index = i + 1
            elif stripped == '' and insert_index > 0:
                insert_index = i + 1
            elif stripped != '' and not stripped.startswith('//'):
                break

        # Insert @use statement
        lines.insert(insert_index, "@use '@/styles' as *;")
        if insert_index < len(lines) - 1 and lines[insert_index + 1].strip() != '':
            lines.insert(insert_index + 1, '')

        content = '\n'.join(lines)

        # Clean up multiple blank lines
        content = re.sub(r'\n\n\n+', '\n\n', content)

        # Write back
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)

        migrated_count += 1
        print(f"+ Migrated: {os.path.basename(file_path)}")

print(f"\n* Migration complete! Updated {migrated_count} files.")
print("\n- Next steps:")
print("1. Run: cd monte-da-estrada && npm run dev")
print("2. Verify no SCSS errors in console")
